### ISSUE TITLE
Adding Suspendable Functional Interfaces

### ISSUE THREAD

**Issue by MatanSabag** [puniverse/quasar#259]
Java's built-in functional interfaces (such as Consumer) don't support Fibers because they don't throw SuspendableExecution. 

Adding a library for that is needed. 
In my own projects, I added some (such as SuspendableConsumer), and I can find time to work on it and contribute it if you wish.

Matan

**Comment by pron** [comment]
Sure, but as a separate project first. Just some interface definitions, right?

**Comment by MatanSabag** [comment]
Yes, exactly, copying the same interface definitions with the appropriate Suspendable  modification.
Should I open a github project for that and send a link to you here when it's done?

**Comment by pron** [comment]
Yes, please.
